google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -182,7 +183,9 @@ public void customize(JList list, Artifact value, int index, boolean selected,
     return ArtifactUtil.getModulesIncludedInArtifacts(Collections.singletonList(artifact), project)
         .stream()
         .map(AppEngineStandardFacet::getAppEngineFacetByModule)
-        .findFirst();
+        .map(Optional::ofNullable)
+        .findFirst()
+        .flatMap(Function.identity());
```

Comment:
the flatMap? its to flatten out the Optional created by :186

`map(Optional::ofNullable)` will create create a stream of Optionals (replacing nulls with Optional.empty). findFirst will then return the first Optional (in our error scenario, it will be Optional.empty). But since findFirst itself wraps the result in an Optional, it will be: `Optional<Optional>`. So we need to flatten it out.